File: src/test/java/com/leetcode/P318Test.java
package com.leetcode;

import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

class P318Test {
    private P318 sut = new P318();

    @Test
    void example01() {
        assertThat(sut.maxProduct(new String[]{"abcw", "baz", "foo", "bar", "xtfn", "abcdef"}), is(16));
    }

    @Test
    void example02() {
        assertThat(sut.maxProduct(new String[]{"a", "ab", "abc", "d", "cd", "bcd", "abcd"}), is(4));
    }

    @Test
    void example03() {
        assertThat(sut.maxProduct(new String[]{"a", "aa", "aaa", "aaaa"}), is(0));
    }

}

File: src/test/kotlin/com/leetcode/P318Test.kt
package com.leetcode

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.jupiter.api.Test

class P318Test {
    val sut = P318()

    @Test
    fun `example 1`() {
        assertThat(
            sut.maxProduct(arrayOf("abcw", "baz", "foo", "bar", "xtfn", "abcdef")),
            `is`(16)
        )
    }

    @Test
    fun `example 2`() {
        assertThat(
            sut.maxProduct(arrayOf("a", "ab", "abc", "d", "cd", "bcd", "abcd")),
            `is`(4)
        )
    }

    @Test
    fun `example 3`() {
        assertThat(
            sut.maxProduct(arrayOf("a", "aa", "aaa", "aaaa")),
            `is`(0)
        )
    }
}